File: AnimaLost/AnimaLost/Model/Species.cs
using System.Collections.Generic;

namespace AnimaLost.Model
{
    public class Species
    {
        public Species()
        {
            Breed = new HashSet<Breed>();
        }

        public int Id { get; set; }
        public string Name { get; set; }

        public ICollection<Breed> Breed { get; set; }
    }
}

File: SmartCityAndroid/app/src/main/java/com/nicolas/smartcityandroid/Model/Species.java
package com.nicolas.smartcityandroid.Model;

import android.support.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.ArrayList;

public class Species implements Serializable {
    @SerializedName("name")
    @NonNull
    private String name;
    @SerializedName("breed")
    private ArrayList<Breed> breed;

    @NonNull
    public String getName() {
        return name;
    }

    public void setName(@NonNull String name) {
        this.name = name;
    }

    public ArrayList<Breed> getBreed() {
        return breed;
    }

    public void setBreed(ArrayList<Breed> breed) {
        this.breed = breed;
    }

    public Species(@NonNull String name, ArrayList<Breed> breed) {
        setName(name);
        setBreed(breed);
    }
}
